pandas/tests/io/parser/test_na_values.py

```diff
@@ -146,20 +148,77 @@ def test_custom_na_values(all_parsers, na_values):
     tm.assert_frame_equal(result, expected)
 
 
-def test_bool_na_values(all_parsers):
-    data = """A,B,C
-True,False,True
-NA,True,False
-False,NA,True"""
+@pytest.mark.parametrize(
+    "use_nullable_dtypes, expected",
+    [
+        (
+            True,
+            DataFrame(
+                {
+                    "A": pd_array([True, NA, False], dtype="boolean"),
+                    "B": pd_array([False, True, NA], dtype="boolean"),
+                    "C": np.array([np.nan, np.nan, np.nan], dtype="float64"),
+                    "D": np.array([True, False, True], dtype="bool"),
```

Comment:
We should probably use "boolean" dtype if `use_nullable_dtypes=True` regardless of whether NA's where actually present or not?

Reply:
All NAs has unknown dtype, so we follow numpy and return float64. Currently, there has to be an integer or boolean value and NAs in order to return a nullable EA type, so a column of a float or bool will just return the numpy types.